File: hola_cliente.py
import sys
sys.path.append("gen-py")
from hola import HolaServicio

from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol

transporte = TSocket.TSocket("localhost", 9090)
transporte = TTransport.TBufferedTransport(transporte)
protocolo = TBinaryProtocol.TBinaryProtocol(transporte)
cliente = HolaServicio.Client(protocolo)

transporte.open()
mensaje = cliente.hola_func()
print("[Cliente] Se recibio %s" % mensaje)
transporte.close()

File: HolaCliente.java
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.TException;

public class HolaCliente {
    public static void main(String[] args) {
        TSocket trans = new TSocket("localhost", 9090);
        TBinaryProtocol protocolo = new TBinaryProtocol(trans);
        HolaServicio.Client cliente = new HolaServicio.Client(protocolo);
        
        try {
            trans.open();
            String cadena = cliente.hola_func();
            System.out.println("[Cliente] recibido: " + cadena);
        } catch (TException ex) {
            ex.printStackTrace();
        }
        trans.close();
    }
}